src/DynamicData/List/IChangeSet.cs

```diff
@@ -24,5 +24,7 @@ public interface IChangeSet<TObject> : IEnumerable<Change<TObject>>, IChangeSet
         /// The total count of items changed
         /// </summary>
         int TotalChanges { get; }
+
```

Comment:
I see why you have added the equality comparer to the change set.  However,  I do not see this as being the correct solution to pass the equality comparer into the clone method.  The change set exists for the purpose of transmitting changes only with a very well defined API. To add this new property to the public API for the purpose of a single function is overkill as it is irrelevant for the rest of the system.

I am sure there is a functional way of passing the comparer. I have grabbed your branch and will take a look to see if I can find an obvious alternative.

Reply:
I understand. That's exactly why I didn't go too far in this PR.

Reply:
PR updated with your suggestion